docs/proposals/job.md

```diff
@@ -0,0 +1,145 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller - which will be responsible
+for managing pod(s) that require to run-once to a completion, in contrast to what
+ReplicationController currently offers.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+* New Job resource [#7380](https://github.com/GoogleCloudPlatform/kubernetes/pull/7380)
+
+
+## Use Cases
+1. Be able to start of one or several pods tracked as a single entity.
+1. Be able to implement heavy-computation model, eg. MapReduce.
+1. Be able to get the job status.
+1. Be able to limit the execution time for a job.
+1. Be able to create a chain of jobs dependent one on another.
```

Comment:
Makes sense - you can implement dependencies in a single pod if necessary.

On Wed, Jul 22, 2015 at 1:35 PM, Brian Grant <notifications@github.com>
wrote:

> In docs/proposals/job.md
> <https://github.com/openshift/origin/pull/3693#discussion_r35240307>:
>
> > +## Abstract
> > +A proposal for implementing a new controller - Job controller - which will be responsible
> > +for managing pod(s) that require to run-once to a completion, in contrast to what
> > +ReplicationController currently offers.
> > +
> > +Several existing issues and PRs were already created regarding that particular subject:
> > +* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
> > +* New Job resource [#7380](https://github.com/GoogleCloudPlatform/kubernetes/pull/7380)
> > +
> > +
> > +## Use Cases
> > +1. Be able to start of one or several pods tracked as a single entity.
> > +1. Be able to implement heavy-computation model, eg. MapReduce.
> > +1. Be able to get the job status.
> > +1. Be able to limit the execution time for a job.
> > +1. Be able to create a chain of jobs dependent one on another.
>
> Run-after dependencies are a tricky area:
>
>    - need a convenient mechanism for creating references to other jobs
>    - need to ensure completion status is not lost until it is consumed by
>    dependencies
>    - need to kill dependent jobs in the case of failure, with clear
>    reason attribution
>    - eventually need to support cross-cluster dependencies
>
> I recommend deferring this feature to later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/3693/files#r35240307>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift